Server/src/services/getBasicDataService.ts

```diff
@@ -0,0 +1,28 @@
+import { Connection, getConnection } from "typeorm";
+import { selectAllCategoriesQuery } from "../models/entities/Category";
+import { selectAllMaterialsQuery } from "../models/entities/Material";
+import { selectAllSubcategoriesQuery } from "../models/entities/Subcategory";
+import { ICategory } from "../models/interfaces/CategoryInterface";
+import { IMaterial } from "../models/interfaces/MaterialInterface";
+import { ISubcategory } from "../models/interfaces/SubcategoryInterface";
+
+export class retrieveBasicData {
+    private connection: Connection;
+    constructor() {
+        this.connection = getConnection();
+    }
+
+    async getBasicCategoryData() {
+        let categoryData: ICategory[] = await selectAllCategoriesQuery(this.connection.manager)

```

Comment:
Is there any chance something here will fail? The controller has wrapped this call in a try catch block. Where would the error come from?

Reply:
I was under the mistaken impression that this was proper practice to wrap requests in a try/catch block. I've removed it.

Reply:
This is still needed

Reply:
I'm confused, is the implication that the try/catch block is in fact needed?